src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -965,9 +981,6 @@ export class DashboardAppController {
       if (outputSubscription) {
         outputSubscription.unsubscribe();
       }
-      if (dashboardContainer) {
-        dashboardContainer.destroy();
```

Comment:
@majagrubic, sorry for following up on merged pr. 
I wonder if this was moved some where or called somewhere else implicitly? 

Otherwise it seems like moving between dashboards or navigating between dashboard / discover / visualize without full page reload would cause memory leaks and probably other undesired effects, since subscriptions within container are not cleaned up?

Reply:
I put up this tiny pr: https://github.com/elastic/kibana/pull/64154
hope this should address underlying issue. Will see if tests pass 🙏